applications/UMC-Tokenscribe.md

```diff
@@ -0,0 +1,159 @@
+This application is aimed to propose a specification of secured token subscription which is the key feature of UMC - Universal Membership Community.
+
+#### Integration
+
+The secured token subscription will be implemented with Ink!.
+
+#### Team Interest
+
+We are dapp developers and we want to design a subscription feature for our project to manage users' membership with the secured token subscription. Our project is aimed to build a universal membership network which is can be used for both on-chain and off-chain services, such as Shopping Mall Memberships.
+
+### Project Details 
+
+With EIP1337, all the subscription status-related information is managed by a subscription contract, which will expose the users to the risks just like the situation of subscription in the telecom age and the early age of the internet. Most scamming cases are let users subscribe to some uncertain subscriptions while users cannot cancel those subscriptions due easily.
+The main idea is to track and manage the subscription in token contract rather than subscription contracts to achieve the goal of securing users' assets. 
+
+
+#### Interface Specification
+
+The standard methods of ERC20 should be implemented as part of the token with the subscription feature.
+
+The following methods also should be implemented.
+
+```
+subscribe
+
+- desc: subscribe the specific service.
+- params: contract address, interval, extra data
+- return: success
+```
+
+```
+unsubscribe
+
+- desc: unsubscribe the specific service.
+- params: contract address, data
+- return: success
+```
+
+```
+subscriptionStatus
+
+- desc: return the status of specific subscription.
+- params: user address, contract address
+- return: the status of subscription
+```
+
+```
+subscriptionInterval
+
+- desc: return the interval of specific subscription.
+- params: user address, contract address
+- return: the interval of subscription
+```
+
+```
+executeSubscription
+
+- desc: execute the subscription for a specific address if pass the check.
+- params: user address
+- return: success
+```
+
+```
+validSubscriptions
+
+- desc: return the list of current valid subscriptions.
+- params: user address
+- return: the list of current valid subscriptions
+```
+
+### Ecosystem Fit 
+
+[EIP1337 Draft](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1337.md) is the only one proposed in EIPs for token subscription, but still not widly used since approved six monthes before.
+
+[ERC20 with Ink!](https://github.com/paritytech/ink/blob/master/examples/erc20/lib.rs) is the sample code of ERC20 token in Ink!.
+
+At the very beginning stage of Ink! and Polkadot ecosystem, it's good to add token subscription as the common interface for tokens.
```

Comment:
Have you looked at https://github.com/w3f/PSPs/ in regards to token interfaces?

Reply:
@lamafab  Yes, I have looked at PSPs repo as mentioned by Noc2. As I understand, PSPs is the repo for all standard related specifications. My idea is to define token subscription in the interface of tokens.

Reply:
Yes, I have looked at PSPs repo as mentioned by Noc2. As I understand, PSPs is the repo for all standard related specifications. My idea is to define token subscription in the interface of tokens.